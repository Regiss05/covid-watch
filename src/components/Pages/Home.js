// @ts-nocheck
import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import CountriesList from '../Country/CountriesList';
// import africaImage from '../../assets/africa.png';
import classes from './Home.module.css';

const Home = () => {
  const { countries } = useSelector((state) => state.countries);
  const [searchValue, setSearchValue] = useState('');

  const totalVacinated = () => {
    let total = 0;
    if (countries) {
      countries.forEach((country) => {
        total += country.total;
      });
    }
    return total.toLocaleString();
  };

  const fullyVacinated = () => {
    let total = 0;
    if (countries) {
      countries.forEach((country) => {
        total += country.full_vacinated;
      });
    }
    return total.toLocaleString();
  };

  const partiallyVacinated = () => {
    let total = 0;
    if (countries) {
      countries.forEach((country) => {
        total += country.partial_vaccinated;
      });
    }
    return total.toLocaleString();
  };
  const filterCountriesHandler = (e) => {
    setSearchValue(e.target.value);
  };

  return (
    <>
      <section className={classes.homeCallout}>
        <div className={classes.innercontent}>
          <h1>Africa Vaccine Stats</h1>
          <p>
            Total Vaccines Administed:
            {' '}
            {totalVacinated()}
          </p>
          <p>
            Fully Vaccinated Individuals:
            {' '}
            {fullyVacinated()}
          </p>
          <p>
            Partially Vaccinated Individuals:
            {' '}
            {partiallyVacinated()}
          </p>
        </div>
      </section>
      <section className={classes.dataSection}>
        <h2 className={classes.dataSectionHeader}>
          Africa COVID Vaccine Data
        </h2>
        <div className={classes.inputDiv}>
          <label htmlFor="Search" className={`${classes.field} ${classes.field_v1}`}>
            <input
              className={`${classes.field__input} ${classes.field__inputs}`}
              type="text"
              name="search"
              value={searchValue}
              onChange={filterCountriesHandler}
              placeholder="e.g. Kenya"
            />
            <span className={`${classes['field__label-wrap']}`}>
              <span className={`${classes.field__label}`}>Search Country</span>
            </span>
          </label>
        </div>

        <div className={`${classes.homeContent}`}>
          <CountriesList searchValue={searchValue} />
        </div>
      </section>
    </>
  );
};

export default Home;
