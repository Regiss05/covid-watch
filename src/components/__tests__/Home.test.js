import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import Home from '../Pages/Home';
import store from '../../redux/configureStore';

describe('Home tests', () => {
  test('should render Home compponent', () => {
    const homePage = render(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
        ,
      </Provider>,
    );

    expect(homePage).toMatchSnapshot();
  });
});
