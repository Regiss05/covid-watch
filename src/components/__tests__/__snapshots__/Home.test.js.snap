// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home tests should render Home compponent 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="homeCallout"
      >
        <div
          class="africaImageDiv"
        >
          <img
            alt="Map of Africa"
            class="africaImage"
            src="africa.png"
          />
        </div>
        <div
          class="innercontent"
        >
          <h1>
            Africa Vaccine Stats
          </h1>
          <p>
            Total Vaccines Administed:
             
            0
          </p>
          <p>
            Fully Vaccinated Individuals:
             
            0
          </p>
          <p>
            Partially Vaccinated Individuals:
             
            0
          </p>
        </div>
      </section>
      <section
        class="dataSection"
      >
        <h2
          class="dataSectionHeader"
        >
          Africa COVID Vaccine Data
        </h2>
        <div
          class="inputDiv"
        >
          <label
            class="field field_v1"
            for="Search"
          >
            <input
              class="field__input field__inputs"
              name="search"
              placeholder="e.g. Kenya"
              type="text"
              value=""
            />
            <span
              class="field__label-wrap"
            >
              <span
                class="field__label"
              >
                Search Country
              </span>
            </span>
          </label>
        </div>
        <div
          class="homeContent"
        >
          <div
            class="countryWrapper"
          />
        </div>
      </section>
      ,
    </div>
  </body>,
  "container": <div>
    <section
      class="homeCallout"
    >
      <div
        class="africaImageDiv"
      >
        <img
          alt="Map of Africa"
          class="africaImage"
          src="africa.png"
        />
      </div>
      <div
        class="innercontent"
      >
        <h1>
          Africa Vaccine Stats
        </h1>
        <p>
          Total Vaccines Administed:
           
          0
        </p>
        <p>
          Fully Vaccinated Individuals:
           
          0
        </p>
        <p>
          Partially Vaccinated Individuals:
           
          0
        </p>
      </div>
    </section>
    <section
      class="dataSection"
    >
      <h2
        class="dataSectionHeader"
      >
        Africa COVID Vaccine Data
      </h2>
      <div
        class="inputDiv"
      >
        <label
          class="field field_v1"
          for="Search"
        >
          <input
            class="field__input field__inputs"
            name="search"
            placeholder="e.g. Kenya"
            type="text"
            value=""
          />
          <span
            class="field__label-wrap"
          >
            <span
              class="field__label"
            >
              Search Country
            </span>
          </span>
        </label>
      </div>
      <div
        class="homeContent"
      >
        <div
          class="countryWrapper"
        />
      </div>
    </section>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
